RELEASE_NOTES.md

```diff
@@ -10,6 +10,51 @@
 
 - The battery pool metric methods no longer return `None` when no batteries are available. Instead, the value of the `Sample` or `PowerMetric` is set to `None`.
 
+- The power distribution `Result` is now a union of all different types of results rather than a base class. This means we can now also use `match` to check for result types instead of only using `isinstance()`. The following example shows how `Result` can be used for matching power distribution results:
+
+  ```python
+    def handle_power_request_result(result: Result) -> None:
+        match result:
+            case Success() as success:
+                print(f"Power request was successful: {success}")
+            case PartialFailure() as partial_failure:
+                print(f"Power request was partially successful: {partial_failure}")
+            case OutOfBounds() as out_of_bounds:
+                print(f"Power request was out of bounds: {out_of_bounds}")
+            case Error() as error:
+                print(f"Power request failed: {error}")
+            case _:
+                assert False, "Should not reach this point"
+
+    request = Request(
+        namespace="TestChannel",
+        power=Power.from_watts(123.4),
+        batteries={8, 18},
+    )
+
+    results: list[Result] = [
+        Success(
+            request,
+            succeeded_power=Power.from_watts(123.4),
+            succeeded_batteries={8, 18},
+            excess_power=Power.zero(),
+        ),
+        PartialFailure(
+            request,
+            succeeded_power=Power.from_watts(103.4),
+            succeeded_batteries={8},
+            excess_power=Power.zero(),
+            failed_batteries={18},
+            failed_power=Power.from_watts(20.0),
+        ),
+        OutOfBounds(request, bounds=PowerBounds(0, 0, 0, 800)),
+        Error(request, msg="The batteries are not available"),
+    ]
+
+    for r in results:
+        handle_power_request_result(r)
+  ```
```

Comment:
I think this might be too much for a release note, I think this should suffice:

~~~suggestion
  ```python
  from typing import assert_never
  result: Result = some_operation()
  match result:
      case Success() as success:
          print(f"Power request was successful: {success}")
      case PartialFailure() as partial_failure:
          print(f"Power request was partially successful: {partial_failure}")
      case OutOfBounds() as out_of_bounds:
          print(f"Power request was out of bounds: {out_of_bounds}")
      case Error() as error:
          print(f"Power request failed: {error}")
      case _ as unreachable:
          assert_never(unreachable)
  ```
~~~

Using `assert_never` makes sure that if a new result is added, the `match` is forced to handle it (otherwise it will be a `mypy` error).

Reply:
right, that was indeed too much :angel: Thanks for the suggestions!